### DISCUSSION TITLE
Releases

### DISCUSSION THREAD

**Discussion by dishwasher-detergent** [sanny-io/react-appwrite#33] [General]
At what point are you planning on making another release? @sanny-io 

Also what quantifies a new release?

**Comment by sanny-io** [comment]
There's been enough changed that it's time for a new one. I wonder what version it should be? Probably not 1.0.0 because we don't have all of the endpoints and services supported yet. What do you guys think about release frequency and versioning? I don't have anything set in mind. We can establish how we want to manage this moving forward now.

@dishwasher-detergent 
@probablyArth 
@safwanyp

**Reply by probablyArth** [reply]
probably just increment 0.x until we have proper tests and all of the services setup in a robust and non-breaking way.
Current major TODOs that I can think of before we release v1:
- optional realtime for all, its just there for some and not for some lol
- cover all the endpoints for all the services
- write tests, I have no idea how that's going to happen

Moving forward, we should probably add major releases only when there's a new feature and minor releases for bug fixes?

**Reply by safwanyp** [reply]
I was thinking we could follow this convention until we support all endpoints, along with realtime (as @probablyArth) mentioned:
- For each group of endpoints that we implement (e.g. Teams, Accounts, Databases, Storage, etc.), we could increment by 0.x
- For each individual endpoint in a group, we could increment by 0.0.x

Once we reach 100% support, we should work on a more interactive example project, test all endpoints, and release v1.

Once we have released v1, we could scrap this convention, and work on a new one to keep up with Appwrite's updates.

**Reply by sanny-io** [reply]
@probablyArth do you mean minor releases and patch releases? Major release would be 1.0.0

| 1     | 0     | 0     | 
|-------|-------|-------|
| major | minor | patch | 

Yeah, the more I think about it, there is no way for us to realistically follow semver at this time considering backwards incompatible changes are going to be necessary relatively frequently until things are stabilized.

A minor release for the completion of a service and a patch release for new features/bug fixes sounds good. I am in agreement with you guys there. Then as @safwanyp said we can scrap that later.

I will publish 0.4.0 this weekend if there are no objections to that version, and will continue publishing releases every week or 2 according to this convention.
